tasks/db_deploy/db_deploy.py

```diff
@@ -6,376 +6,424 @@
+               drop_database: bool) -> str:  # todo: Once large tests are adjusted/dropped, consider code reorg.
+    """
+    Task called by the handler to perform the work.
+
+    Args:
+        db_host: The database host.
+        db_name: The name of the database.
+        db_port: The database port.
+        db_user: The name of the application user.
+        db_user_pass: The password for the application user to connect to the database with.
+        db_admin_pass: An admin level password to connect to the database with.
+        drop_database: When true, will execute a DROP DATABASE command.
+
+    Returns:
+        string: description of status.
+
+    Raises:
+        DatabaseError: An error occurred.
+    """
+    log_status("start")
+
+    # connect as postgres to create the new database
+    con = get_db_connection(db_host, 'postgres', db_port, 'postgres', db_admin_pass)
+
+    log_status("Connected to postgres")
+    db_existed, status = create_database(con, drop_database)
     if not db_existed:
-        os.environ["DATABASE_PW"] = db_pw
-        status = create_roles_and_users(con, db_user)
+        create_roles_and_users(con, db_user, db_user_pass)
     con.close()
 
-    #connect to the database we just created as postgres
-    os.environ["DATABASE_NAME"] = db_name
-    os.environ["DATABASE_PW"] = master_user_pw
-    con = get_db_connnection()
-    status = log_status(f"connected to {db_name}")
+    # Connect to the database we just created.
+    con = get_db_connection(db_host, db_name, db_port, 'postgres', db_admin_pass)
+    log_status(f"connected to {db_name}")
     status = create_schema(con)
     con.close()
-    status = create_tables()
+    create_tables(db_host, db_name, db_port, 'postgres', db_admin_pass)
 
-    status = log_status("database ddl execution complete")
+    log_status("database ddl execution complete")
     return status
 
-def create_database(con):
+
+def create_database(con: connection, drop_database: bool) -> Tuple[bool, str]:
     """
     Creates the database, dropping it first if requested.
 
-        Args:
-            con (object): a connection to the postgres database
+    Args:
+        con: A connection to the postgres database.
+        drop_database: When true, will execute a DROP DATABASE command.
 
-        Returns:
-            bool: True if the database already existed
-                when the create was run.
-            string: description of status of create database.
+    Returns:
+        bool: True if the database already existed
+            when the create was run.
+        string: description of status of create database.
 
-        Raises:
-            DatabaseError: An error occurred.
+    Raises:
+        DatabaseError: An error occurred.
     """
     con.set_isolation_level(ISOLATION_LEVEL_AUTOCOMMIT)
     cur = get_cursor(con)
-    try:
-        drop = os.environ["DROP_DATABASE"]
-    except KeyError:
-        drop = "False"
-    if drop == "True":
+
+    if drop_database:
```

Comment:
Think about a log warn here stating database will be dropped.

Reply:
Added